2011-03-18  Christian Bruel  <christian.bruel@st.com>

	* sh.md (movsf_ie): Don't use R0 for float const reload.

--- gcc-4.5.2.orig/gcc/config/sh/sh.md	2011-03-16 14:07:33.000000000 +0100
+++ gcc-4.5.2/gcc/config/sh/sh.md	2011-03-16 14:10:57.000000000 +0100
@@ -6640,7 +6640,7 @@
 	(match_operand:SF 1 "general_movsrc_operand"
 	  "f,r,G,H,FQ,mf,f,FQ,mr,r,y,f,>,fr,y,r,y,>,y"))
    (use (match_operand:PSI 2 "fpscr_operand" "c,c,c,c,c,c,c,c,c,c,c,c,c,c,c,c,c,c,c"))
-   (clobber (match_scratch:SI 3 "=X,X,Bsc,Bsc,&z,X,X,&z,X,X,X,X,X,y,X,X,X,X,X"))]
+   (clobber (match_scratch:SI 3 "=X,X,Bsc,Bsc,&z,X,X,X,X,X,X,X,X,y,X,X,X,X,X"))]
 
   "TARGET_SH2E
    && (arith_reg_operand (operands[0], SFmode)
